### ISSUE TITLE
Map that have a lots of 'fake file'

### ISSUE THREAD

**Issue by KrayOristine** [ladislav-zezula/StormLib#363]
I have just encountered this 'fabulous' map that use 'new protection' methods. At first glance, it seem like it do not modify the MPQ Header (still the usual 77, 80, 81, 26)

But as a closer look by opening it on MPQ Editor, the editor 'recognize' **30,000+ files!!**

As i navigate through the MPQ tables, i saw file that have 0 byte of data?? and there is not just one, there is a lot of them have 0 byte of actual data

So based on these 'evidence' i assume this 'protection method' add info about file that doesn't exists to confuse the mpq editor, and probably because the editor have to handle like 10,000+ files so it some time just crashed when i navigating through those fake dummy file.

The map i mentioned is uploaded here (can't attach because it is larger than 25MB): https://pixeldrain.com/u/R41YzpCN

**Comment by ladislav-zezula** [comment]
Yeah, having a lot of fake files is a popular method. Not much I can do. MPQ Editor can do `Deep File Verification` - during open, it tries to read every file (or at least part of it) and the resulting view will only contain files that are successfully verified. When I tried on this map, I only got like 6100 files in the root folder of the map.

**Comment by KrayOristine** [comment]
> Yeah, having a lot of fake files is a popular method. Not much I can do. MPQ Editor can do `Deep File Verification` - during open, it tries to read every file (or at least part of it) and the resulting view will only contain files that are successfully verified. When I tried on this map, I only got like 6100 files in the root folder of the map.

The thing i worry about is it have file that have data from random sector,  can v4.0.0.924 handle that?

**Comment by ladislav-zezula** [comment]
> So can that issue be some how 'improved'?

No. By the nature of MPQ archives, an arbitraty of files can point to a single data.

**Comment by KrayOristine** [comment]
> No. By the nature of MPQ archives, an arbitraty of files can point to a single data.

Damn, okay then. I will close this issue because it cant be 'fixed'
